src/main/scala/mesosphere/util/JavaScalaFutureSupport.scala

```diff
@@ -0,0 +1,31 @@
+package mesosphere.util
+
+import java.util.concurrent.{ Callable, Future => JFuture, FutureTask }
+
+import scala.concurrent.duration._
+import scala.concurrent.{ Await, Future }
+
+/**
+  * Support for conversion between Java and Scala futures.
+  * TODO merge to BackToTheFuture in mesosphere.utils
+  */
+trait JavaScalaFutureSupport {
+
+  /**
+    * Converts a Scala Future to a Java Future
+    */
+  implicit class Scala2JavaFuture[A](f: Future[A]) {
+
+    def asJava(duration: Duration): JFuture[A] = {
+      val task = new FutureTask[A](new Callable[A] {
+        override def call(): A = Await.result(f, duration)
+      })
+      new Thread(task).run()
```

Comment:
This is very expensive. Please instead wrap the `scala.concurrent.Future` in a class that implements `java.util.concurrent.Future` and delegates the methods.